microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -20,30 +19,305 @@
+                # 18.04 doesn't need linux-modules-extra-azure
+                # since it will never have MANA support
+            ],
+            exclusive_match=True,
+        ),
+        OsPackageDependencies(
+            matcher=lambda x: isinstance(x, Debian),
+            packages=[
+                "build-essential",
+                "libnuma-dev",
+                "libmnl-dev",
+                "python3-pyelftools",
+                "libelf-dev",
+                "pkg-config",
+                "linux-modules-extra-azure",
+            ],
+            exclusive_match=True,
+        ),
+        OsPackageDependencies(
+            matcher=lambda x: isinstance(x, Suse),
+            packages=[
+                "psmisc",
+                "libnuma-devel",
+                "numactl",
+                "libmnl-devel meson",
+                "gcc-c++",
+            ],
+            exclusive_match=True,
+        ),
+        OsPackageDependencies(
+            matcher=lambda x: isinstance(x, (Fedora)),
+            packages=[
+                "psmisc",
+                "numactl-devel",
+                "pkgconfig",
+                "elfutils-libelf-devel",
+                "python3-pip",
+                "kernel-modules-extra",
+                "kernel-headers",
+                "gcc-c++",
+            ],
+            exclusive_match=True,
+        ),
+        OsPackageDependencies(matcher=invalid_os_thrower),
+    ]
+)
+
+
+def get_debian_backport_repo_args(os: Debian) -> List[str]:
+    if not isinstance(os, Debian):
+        return []
+    if isinstance(os, Ubuntu) and (
+        is_ubuntu_latest_or_prerelease(os) or not is_ubuntu_lts_version(os)
+    ):
+        return []
+    repos = os.get_repositories()
+    backport_repo = f"{os.information.codename}-backports"
+    if any([backport_repo in repo.name for repo in repos]):
+        return [f"-t {backport_repo}"]
+    return []
+
+
+class DpdkPackageManagerInstall(PackageManagerInstall):
+    def _setup_node(self) -> None:
+        if isinstance(self._os, Debian):
+            self._package_manager_extra_args = get_debian_backport_repo_args(self._os)
+            if self._os.information.version < "22.4.0":
+                self._os.update_packages("linux-azure")
+                self._node.reboot()
+        elif isinstance(self._os, Fedora):
+            self._os.install_epel()
+
+    def get_installed_version(self) -> VersionInfo:
+        return self._os.get_package_information("dpdk", use_cached=False)
+
+    def _check_if_installed(self) -> bool:
+        return self._os.package_exists("dpdk")
+
+
+# implement SourceInstall for DPDK
+class DpdkSourceInstall(Installer):
+    _sample_applications = [
+        "l3fwd",
+        "multi_process/client_server_mp/mp_server",
+        "multi_process/client_server_mp/mp_client",
+    ]
+
+    def _setup_installation(self) -> None:
+        super()._setup_installation()
+        self._source_path = self._asset_path
```

Comment:
If the `_source_path` is the same as `_asset_path`, does it need an extra `_source_path` variable?

Reply:
only if you want the variable name to make sense. I'll change